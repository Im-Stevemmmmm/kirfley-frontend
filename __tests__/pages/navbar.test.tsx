import { act, render, screen } from '@testing-library/react';
import ReactDOM from 'react-dom';
import IndexNavbar from '../../components/index-navbar';

it('renders correctly', () => {
  const container = document.createElement('div');

  act(() => {
    ReactDOM.render(<IndexNavbar />, container);
  });

  expect(container).toMatchSnapshot();
});

describe('navbar', () => {
  beforeEach(() => render(<IndexNavbar />));

  it('renders the sign up and log in buttons', () => {
    expect(screen.getByRole('button', { name: 'Sign Up' })).toBeInTheDocument;
    expect(screen.getByRole('button', { name: 'Log In' })).toBeInTheDocument;
  });
});
