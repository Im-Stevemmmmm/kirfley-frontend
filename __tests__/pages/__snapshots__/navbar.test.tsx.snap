// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    id="navbar"
  >
    <ul
      id="navbarNav"
    >
      <li>
        <div
          id="logo"
        >
          <div
            style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
          >
            <div
              style="box-sizing: border-box; display: block; max-width: 100%;"
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,<svg width=\\"50\\" height=\\"50\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                style="max-width: 100%; display: block;"
              />
            </div>
            <img
              alt=""
              decoding="async"
              src="/_next/image?url=%2Ffern.svg&w=320&q=75"
              srcset="/_next/image?url=%2Ffern.svg&w=320&q=75 1x"
              style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
            />
          </div>
        </div>
      </li>
      <li>
        <button
          class="standard"
        >
          Log In
        </button>
      </li>
      <li>
        <button
          class="inverted"
        >
          Sign Up
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
